### PR TITLE
Add necessary face identification files/directories [GG-43]

### PR DESCRIPTION
**Pull request by kemptonk**
Note: The programs may not actually function, they files are just there to indicate the directory structure.

### CODE REVIEW

File: FaceID/TestCode/camTest.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    for imagePath in imagePaths:
+        PIL_img = Image.open(imagePath).convert('L') # convert it to grayscale
+        img_numpy = np.array(PIL_img,'uint8')
+        id = int(os.path.split(imagePath)[-1].split(".")[1])
+        faces = detector.detectMultiScale(img_numpy)
+        for (x,y,w,h) in faces:
+            faceSamples.append(img_numpy[y:y+h,x:x+w])
+            ids.append(id)
+    return faceSamples,ids
+print ("\n [INFO] Training faces. It will take a few seconds. Wait ...")
```
Comment (mraynor) [inline FaceID/TestCode/camTest.py]: Can we remove the space between `print` and `(`?

File: FaceID/TestCode/camTest.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import cv2
+import numpy as np
+from PIL import Image
+import os
```
Comment (mraynor) [inline FaceID/TestCode/camTest.py:4]: Typically there are new lines after imports, can we do that here ( and for the other files in this MR)?

File: FaceID/TestCode/camTest.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# Path for face image database
+path = 'dataset'
+recognizer = cv2.face.LBPHFaceRecognizer_create()
+detector = cv2.CascadeClassifier("Cascades/haarcascade_frontalface_default.xml");
+
+# function to get the images and label data
+def getImagesAndLabels(path):
+
+    imagePaths = [os.path.join(path,f) for f in os.listdir(path)]     
+    faceSamples=[]
```
Comment (kesslerk) [inline FaceID/TestCode/camTest.py]: Can you add spaces around the `=`. It makes it more readable and helps with consistency

File: FaceID/TestCode/faceRecogTest.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+cam.set(4, 480) # set video height
+
+# Define min window size to be recognized as a face
+minW = 0.1*cam.get(3)
+minH = 0.1*cam.get(4)
+
+
+while True:
+
+    ret, img =cam.read()
```
Comment (kesslerk) [inline FaceID/TestCode/faceRecogTest.py]: Can you add a space after the `=` for consistency?

File: FaceID/TestCode/faceRecogTest.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        minNeighbors = 5,
+        minSize = (int(minW), int(minH)),
+       )
+   # print(faces) 
+
+
+    for(x,y,w,h) in faces:
+
+        id, confidence = recognizer.predict(gray[y:y+h,x:x+w])
+        if id==1:
```
Comment (kesslerk) [inline FaceID/TestCode/faceRecogTest.py]: Also spaces around the `==` here please

File: FaceID/faceID.py [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+    global photoRegister
+    #photoRegister = readPhotoRegister()
+ 
+    #person = [] # an index inside of the 'photoRegister[]' list
+    #photoNum = len(photoRegister)
+    photoNum = 0 # an index inside of the 'person[]' list
+    frameCount = 50
+
+    flag = False
+    while flag==False:
```
Comment (kesslerk) [inline FaceID/faceID.py]: And spaces around the `==` here as well

File: FaceID/faceID.py [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+
+        empID = input("Enter your EMPLOYEE ID:  ") # get employee ID
+
+        if len(photoRegister)==0:
```
Comment (kesslerk) [inline FaceID/faceID.py]: Same here

File: FaceID/faceID.py [hunk 8]
```diff
@@ -0,0 +1,366 @@
...
+            flag=True
```
Comment (kesslerk) [inline FaceID/faceID.py]: And here. Could you also go through the rest of this file and fix and spacing inconsistency?

File: FaceID/TestCode/faceDetectTest.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+#faceCascade = cv2.CascadeClassifier("/home/pi/opencv-3.3.0/data/haarcascades/home/pi/opencv-3.3.0/data/haarcascades")
+cap = cv2.VideoCapture(0)
+cap.set(3,640) # set Width
+cap.set(4,480) # set Height
+
+
+while True:
+
+    ret, img = cap.read()
+    #	img = cv2.flip(img, -1)
```
Comment (mraynor) [inline FaceID/TestCode/faceDetectTest.py:15]: Do we need this commented line?

File: FaceID/TestCode/faceDetectTest.py [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+    faces = faceCascade.detectMultiScale(
+        gray,
+        scaleFactor = 1.2,
+        minNeighbors = 5,
+        minSize = (20, 20)
+    )
+
+    for (x,y,w,h) in faces:
+        #cv2.rectangle(img,(x,y),(x+w,y+h),(255,0,0),2)
```
Comment (mraynor) [inline FaceID/TestCode/faceDetectTest.py:25]: Do we need this commented line?

**Pull request closed without merging**